src/core/raster/qgsrasterlayer.cpp

```diff
@@ -1646,6 +1646,32 @@ bool QgsRasterLayer::writeXml( QDomNode &layer_node,
   return writeSymbology( layer_node, document, errorMsg, context );
 }
 
+// TODO: this should ideally go to gdal provider (together with most of encodedSource() + decodedSource())
```

Comment:
Hopefully not being too annoying... but what's stopping this? Do we need to make a writeSource/readSource method available to provider libraries?

Reply:
ok, proceed. ;)